packages/clerk-js/src/core/events.ts

```diff
@@ -5,48 +5,45 @@ export const events = {
 } as const;
 
 type ClerkEvent = typeof events[keyof typeof events];
-type EventType<T> = Record<string, unknown> & T;
+type EventHandler<E extends ClerkEvent> = (payload: EventPayload[E]) => void;
 
 type TokenUpdatePayload = { token: TokenResource | null };
 
 type EventPayload = {
-  [events.TokenUpdate]: EventType<TokenUpdatePayload>;
+  [events.TokenUpdate]: TokenUpdatePayload;
 };
 
-class ClerkEventBus extends EventTarget {
-  #handlers = {} as Record<ClerkEvent, Array<any>>;
+const createEventBus = () => {
+  const eventToHandlersMap = new Map<ClerkEvent, Array<EventHandler<any>>>();
 
-  on<E extends ClerkEvent, P extends EventPayload[E]>(eventName: E, callback: (evt: P) => void): void {
-    const handler = (e: Event) => {
-      callback((e as CustomEvent).detail);
-    };
-
-    if (!this.#handlers[eventName]) {
-      this.#handlers[eventName] = [];
+  const on = <E extends ClerkEvent>(event: E, handler: EventHandler<E>) => {
+    if (!eventToHandlersMap.get(event)) {
+      eventToHandlersMap.set(event, []);
     }
-    this.#handlers[eventName].push({ callback, handler });
-
-    this.addEventListener(eventName, handler);
-  }
+    eventToHandlersMap.get(event)?.push(handler);
+  };
 
-  dispatch<E extends ClerkEvent, P extends EventPayload[E]>(eventName: E, data: P): void {
-    this.dispatchEvent(new CustomEvent(eventName, { detail: data }));
-  }
+  const dispatch = <E extends ClerkEvent>(event: E, payload: EventPayload[E]) => {
+    (eventToHandlersMap.get(event) || []).forEach(h => typeof h === 'function' && h(payload));
```

Comment:
❓ Do we need `typeof h === 'function'` since we are using `EventHandler` type ?

Reply:
On the type level, we don't. However, I think these little sanity checks are good practice as an `any`  type on the side of the caller (or the part of the codebase that fires the dispatch call) might crash the host app.